samcli/lib/config/samconfig.py

```diff
@@ -261,6 +244,40 @@ def _deduplicate_global_parameters(self, cmd_name_key, section, key, env=DEFAULT
             # Only keep the global parameter
             del self.document[env][cmd_name_key][section][key]
 
+    @staticmethod
+    def get_default_file(config_dir: str) -> str:
+        """Return a defaultly-named config file, if it exists, otherwise the current default.
+
+        Parameters
+        ----------
+        config_dir: str
+            The name of the directory where the config file is/will be stored.
+
+        Returns
+        -------
+        str
+            The name of the config file found, if it exists. In the case that it does not exist, the default config
+            file name is returned instead.
+        """
+        config_files_found = 0
+        config_file = DEFAULT_CONFIG_FILE_NAME
+
+        for extension in reversed(list(FILE_MANAGER_MAPPER.keys())):
```

Comment:
Q: why reversed?

Reply:
For priority. The code goes through each possible extension, checks if the file exists, and if it does, sets that to be the config file to use. Iterating backwards means that the first extension in the mapper (in this case `.toml`, which is the default) will be used, even if other extensions like `.yaml` are found.

Reply:
While not stated here, it is [left as a comment on the `FILE_MANAGER_MAPPER` itself](https://github.com/aws/aws-sam-cli/blob/e0860ce3615296a17447e868afd521453eefd726/samcli/lib/config/file_manager.py#L337).

Reply:
that dict is ordered by default since python 3.x so that's okay, but the naming does not imply what priority we actually want. Its hidden behind the file manager mapper which makes it less obvious. I would still say make it more explicit.


I approved the PR, but let's look to change it.